server/configHelpers.js

```diff
@@ -0,0 +1,185 @@
+    );
+    fs.mkdirpSync(clientsPath);
+  }
+  fs.writeFileSync(`${clientsPath}/${clientConfig.str('id')}.conf`, configTxt);
+  return clientConfig;
+}
+
+/*
+ * Retrieve a config from clients directory
+ * @param {string} id - id of client to retrieve
+ * @returns {bcfg.Config} - bcfg object of config
+ */
+
+function getConfig(id) {
+  assert(typeof id === 'string', 'Client config must have an id');
+
+  const appConfig = loadConfig('bpanel');
+  const clientsDir = appConfig.str('clients-dir', 'clients');
+
+  const clientsPath = resolve(appConfig.prefix, clientsDir);
+  const config = loadConfig(id, { id, prefix: clientsPath });
+
+  const path = resolve(clientsPath, `${id}.conf`);
+  if (!fs.existsSync(path)) {
+    const error = new Error(`File ${id}.conf not found`);
+    error.code = 'ENOENT';
+    throw error;
+  }
+
+  config.open(`${id}.conf`);
+  return config;
+}
+
+/*
+ * Simple utility that deletes a config
+ * Does not throw errors if client is not found
+ * NOTE: This does not confirm. Deletion is final!
+ * @param {string} id
+ * @returns {bool} - returns true when operation completed successfully
+ */
+
+function deleteConfig(id) {
+  assert(typeof id === 'string', 'Expected to get id of config to delete');
+  const config = getConfig(id);
+  const path = resolve(config.prefix, `${config.str('id')}.conf`);
+  const exists = fs.existsSync(path);
+  if (!exists)
+    logger.info('Attempt to delete config failed because file does not exist');
+  else {
+    try {
+      fs.unlinkSync(path);
+    } catch (e) {
+      logger.error('Problem removing file:', e);
+    }
+  }
+  return true;
+}
+
+/*
+ * create and test clients based on a passed config
+ * @param {Config} clientConfig
+ * @throws {ClientErrors} - throws if at least one client fails
+ * @returns void
+ */
+
+async function testConfigOptions(options) {
+  let clientConfig;
+  try {
+    clientConfig = getConfigFromOptions(options);
+  } catch (e) {
+    throw e;
+  }
+  const { nodeClient, walletClient, multisigWalletClient } = clientFactory(
+    clientConfig
+  );
+  const clientErrors = new ClientErrors();
+
+  try {
+    if (clientConfig.bool('node', true)) await nodeClient.getInfo();
+  } catch (e) {
+    clientErrors.addFailed('node', e);
+  }
+  try {
+    if (clientConfig.bool('wallet', true)) await walletClient.getInfo();
+  } catch (e) {
+    clientErrors.addFailed('wallet', e);
+  }
+  try {
+    if (clientConfig.bool('multisigWallet', true))
+      await multisigWalletClient.getInfo();
+  } catch (e) {
+    clientErrors.addFailed('multisigWallet', e);
+  }
+
+  if (clientErrors.failed.length) {
+    clientErrors.composeMessage();
+    throw clientErrors;
+  }
+}
+
+class ClientErrors extends Error {
+  constructor(...options) {
+    super(...options);
+    this.failed = [];
+  }
+
+  addFailed(clientType, error) {
+    assert(typeof clientType === 'string');
+    this[clientType] = { message: error.message, ...error };
```

Comment:
Is this error object exposed anywhere after being assigned?

Reply:
Sorry, I don't follow.

Reply:
Sorry, could have been more clear. `this[clientType] = { message... }`  <-- I don't see where this is used in the codebase

Reply:
Ah, gotcha. It can be useful in response messages. It's not used right now in the status check, but if you try and POST a new config, without `force`ing it, and it fails, then you'll get a response like this:

```

Reply:
👍